Tests/UnitTests/Core/Propagator/BoundToCurvilinearConversionTests.cpp

```diff
@@ -0,0 +1,602 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/data/test_case.hpp>
+#include <boost/test/unit_test.hpp>
+#include <boost/test/unit_test_log_formatter.hpp>
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/detail/TransformationBoundToFree.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/MagneticField/ConstantBField.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/Propagator/EigenStepper.hpp"
+#include "Acts/Propagator/Navigator.hpp"
+#include "Acts/Propagator/Propagator.hpp"
+#include "Acts/Propagator/StraightLineStepper.hpp"
+#include "Acts/Propagator/VoidNavigator.hpp"
+#include "Acts/Propagator/detail/JacobianEngine.hpp"
+#include "Acts/Surfaces/DiscSurface.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+
+#include <cmath>
+#include <cstdlib>
+#include <exception>
+#include <iomanip>
+#include <iostream>
+#include <typeinfo>
+
+using namespace Acts;
+using namespace Acts::UnitLiterals;
+
+bool isDebugOutputEnabled() {
+  std::array<boost::unit_test::output_format, 1> formats{
+      boost::unit_test::OF_CLF};
+  for (auto a_format : formats) {
+    auto formatter = ::boost::unit_test::unit_test_log.get_formatter(a_format);
+    if (formatter) {
+      return formatter->get_log_level() < boost::unit_test::log_test_units;
+    }
+  }
+  return false;
+}
+
+#define MSG_DEBUG(a)               \
+  if (isDebugOutputEnabled()) {    \
+    std::stringstream msg;         \
+    msg << a;                      \
+    BOOST_TEST_MESSAGE(msg.str()); \
+  }                                \
+  do {                             \
+  } while (0)
```

Comment:
Since it is a boost test, I thought that it makes more sense to use the "logging mechanism" provided by the boost test framework. In this way all the command line arguments should have the desired behavior.